src/pint/utils.py

```diff
@@ -434,3 +435,115 @@ def show_param_cov_matrix(matrix, params, name="Covaraince Matrix", switchRD=Fal
+    If not output save file is desired, save = False (which is the default)
+    Output file name is dmxparse.out
+    """
+    # We get the DMX values, errors, and mjds (same as in getting the DMX values for DMX v. time)
+    # Get number of DMX epochs
+    dmx_epochs = 0
+    for p in fitter.model.params:
+        if "DMX_" in p:
+            dmx_epochs += 1
+    # Get DMX values (will be in units of 10^-3 pc cm^-3)
+    DMX_keys = []
+    DMXs = []
+    DMX_Errs = []
+    DMX_R1 = []
+    DMX_R2 = []
+    DMX_center_MJD = []
+    for ii in range(1, dmx_epochs + 1):
+        if ii < 10:
+            ext = "000%s" % (ii)
+        elif ii >= 10 and ii < 100:
+            ext = "00%s" % (ii)
+        elif ii >= 100 and ii < 1000:
+            ext = "0%s" % (ii)
+        DMX_keys.append("DMX_%s" % (ext))
+        DMXs.append(getattr(fitter.model, "DMX_%s" % (ext)).value)
+        DMX_Errs.append(getattr(fitter.model, "DMX_%s" % (ext)).uncertainty_value)
+        dmxr1 = getattr(fitter.model, "DMXR1_%s" % (ext)).value
+        dmxr2 = getattr(fitter.model, "DMXR2_%s" % (ext)).value
+        DMX_R1.append(dmxr1)
+        DMX_R2.append(dmxr2)
+        DMX_center_MJD.append((dmxr1 + dmxr2) / 2)
+    DMXs = np.array(DMXs)
+    DMX_Errs = np.array(DMX_Errs)
+    DMX_R1 = np.array(DMX_R1)
+    DMX_R2 = np.array(DMX_R2)
+    DMX_center_MJD = np.array(DMX_center_MJD)
+
+    # now get the full parameter covariance matrix from pint
+    # NOTE: we will need to increase all indices by 1 to account for the 'Offset' parameter
+    # that is the first index of the designmatrix
+    fparams = fitter.get_fitparams().keys()
+    params = []
+    for p in fparams:
+        params.append(p)
+    params = np.array(params)
+    p_cov_mat = fitter.covariance_matrix
+    # Now we get the indices that correspond to the DMX values
+    DMX_p_idxs = np.zeros(dmx_epochs, dtype=int)
+    for ii in range(dmx_epochs):
+        DMX_p_idxs[ii] = (
+            int(np.where(params == DMX_keys[ii])[0]) + 1
+        )  # extra 1 is for offset parameters
+    # Sort the array in numerical order for 2.7. 3.5
+    DMX_p_idxs = np.sort(DMX_p_idxs)
+    # Define a matrix that is just the DMX covariances
+    cc = p_cov_mat[
+        DMX_p_idxs[0] : DMX_p_idxs[-1] + 1, DMX_p_idxs[0] : DMX_p_idxs[-1] + 1
+    ]
+    n = len(DMX_Errs)
+    # Find error in mean DM
+    DMX_mean = np.mean(DMXs)
+    DMX_mean_err = np.sqrt(cc.sum()) / float(n)
+    # Do the correction for varying DM
+    m = np.identity(n) - np.ones((n, n)) / float(n)
+    cc = np.dot(np.dot(m, cc), m)
+    DMX_vErrs = np.zeros(n)
+    # We also need to correct for the units here
+    for i in range(n):
+        DMX_vErrs[i] = np.sqrt(cc[i, i])
+    # Check we have the right number of params
+    if len(DMXs) != len(DMX_Errs) or len(DMXs) != len(DMX_vErrs):
+        print("ERROR! Number of DMX entries do not match!")
+        exit()
```

Comment:
I think we should avoid print statements and exits in PINT code.  The following might be better:
```
from astropy import log #  Put at beginning of file
log.error("ERROR! Number of DMX entries do not match!")
raise RuntimeError('Number of DMX entries do not match!')
```